### ISSUE TITLE
Reading parquet file causes a segfault

### ISSUE THREAD

**Issue by fpetkovski** [duckdb/duckdb#6990]
### What happens?

I installed duckdb 0.71 and tried to read a parquet file which lead to a segfault that I cannot troubleshoot/debug.

### To Reproduce

The issue can be reproduced with the following steps. The parquet file is attached as well.

```
$ duckdb                                         
v0.7.1 b00b93f0b1
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
D select count(*) FROM read_parquet('out/data.parquet');
zsh: segmentation fault  duckdb
```

Parquet file: [data.parquet.zip](https://github.com/duckdb/duckdb/files/11170831/data.parquet.zip).

### OS:

OS X, Apple M1 Pro

### DuckDB Version:

0.7.1

### DuckDB Client:

CLI

### Full Name:

Filip Petkovski

### Affiliation:

Shopify

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Tishj** [comment]
```c++
LogicalType ParquetReader::DeriveLogicalType(const SchemaElement &s_ele, bool binary_as_string) {
	// inner node
	D_ASSERT(s_ele.__isset.type && s_ele.num_children == 0);
```
It hits this assertion, type is `INT64` and children is `412`
Sounds like we expect it to not be nested, and it is?

**Comment by fpetkovski** [comment]
Thanks for troubleshooting this. Should DuckDB show an error message instead of exiting with a segfault?  I was able to read the file with the library used to create it and I would not have been able to figure out why DuckDB could not.

**Comment by Tishj** [comment]
I get that it's not very helpful, D_ASSERT is not checked on release (that's why it didn't get caught and resulted in a segfault later on), we might turn it into an if statement that throws an InternalException instead.

@Mytherin you're more familiar with Parquet, is this a corruption or should we just also check if `num_children > 0` to recognize a nested type?

**Comment by Mytherin** [comment]
This file indeed violates the Parquet standard, a node cannot have both `num_children` and `type` set (see [here](https://github.com/apache/parquet-format/blob/2a481fe1aad64ff770e21734533bb7ef5a057dac/src/main/thrift/parquet.thrift#L348)). The file currently essentially says the root element is **both** an integer **and** a struct - where in reality it is a struct. Our Parquet reader would try to read it as an integer, which caused the issue. I assume the source library instead opts to ignore the integer type instead and read it as a struct which is why they can read the file.

Nevertheless, despite the Parquet file being corrupt, we obviously should not crash. I have pushed a fix in #6992 that allows us to read the file by preferring struct types to scalar types, and adding more defensive code that should prevent us from crashing on different corrupt files in this manner.
